parser.js

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env node --abort-on-uncaught-exception
+
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2024 MNX Cloud, Inc.
+ */
+
+
+var child_process = require('child_process');
+var os = require('os');
+var process = require('process');
+
+var VALID_PROTOCOLS = [
+    'tcp',
+    'http',
+    'https'
+];
+
+var MAX_PORT = 65534;
+var MIN_PORT = 1;
+var MAX_BACKENDS_LOW = 32;
+var MAX_BACKENDS_HIGH = 1024;
+var STICKY_MODES = [
+    'http',
+    'https'
+]
+
+var frontends = {};
+var backends  = {};
+
+var dbg = function (x) {
+    console.error(JSON.stringify(x, null, 2));
+};
+
+var mdataGet = function (key, callback) {
+    // assert.string(key, 'key');
+    // assert.func(callback, 'callback');
+
+    var md_cmd = os.platform() === 'sunos' ? '/usr/sbin/mdata-get' :
+        process.cwd() + '/tools/mock-mdata-get';
+
+    // child_process.execFile('/usr/sbin/mdata-get', [
+    child_process.execFile(md_cmd, [
+        key
+    ], function _onMdata(err, stdout, stderr) {
+        // assert.ifError(err, 'mdata-get should always work');
+        if (err || stderr) {
+            // mdata-get error means key was not found. We don't care.
+        }
+        callback(stdout.trim());
+    });
+};
+
+var parseService = function (svcStr, r) {
+    var m = svcStr.match(/([A-z]+):\/\/([0-9]+):(.*)/);
```

Comment:
Does A-z include non-alphabetic characters like [, \, ], ^, _, and `.?  If so, is this intended?

Reply:
No, it's the same as `A-Za-z`.

Going further, the comma `,` is a major token separator. We've already split on that so we'll never get a comma here.

The rest of the characters will return an unparsable chunk.

```
  "ignored": [
    {
      "string": "f[]^_`.?\"'://99999:foo.svc.6d9cb2ba-039c-46f4-94dc-a9fcc34ebb52.us-central-1.cns.mnx.io:99999",
      "errors": [
        "Unparsable chunk"
      ]
    },
```